File: DeckOfCards-Scala/src/main/scala/com/aprearo/deckofcards/PlayWithDeck.scala
package com.aprearo.deckofcards

object PlayWithDeck {
  def main(args: Array[String]) {
    val deck = new Deck()
    println("New deck ready.")
    deck.shuffle
    println("Shuffled deck.")
    Range(0, deck.size).foreach(i => println(s"Dealt: ${deck.dealOne.toString()}. ${i+1} card(s) dealt. ${deck.size} remaining."))
  }
}

File: DeckOfCards-Ruby/src/play_with_deck.rb
require_relative 'deck'
require_relative 'card'

deck = Deck.new
puts "New deck ready."
deck.shuffle
puts "Shuffled deck."
size = deck.size
for i in 0..size-1
    card = deck.deal_one
    puts "Dealt: #{card}. #{i+1} card(s) dealt. #{deck.size} remaining."
end
